src/app/user-list/user-list.component.ts

```diff
@@ -0,0 +1,33 @@
+import { Component, OnInit } from '@angular/core';
+import {UserService} from "../services/user.service";
+import {Router} from "@angular/router";
+import {User} from "../models/User";
+
+@Component({
+  selector: 'app-user-list',
+  templateUrl: './user-list.component.html',
+  styleUrls: ['./user-list.component.scss']
+})
+export class UserListComponent implements OnInit {
+  users: User[] = [];
+
+  constructor(private userService: UserService, private router: Router) { }
+
+  ngOnInit(): void {
+    this.getUsers();
+  }
+
+  getUsers(): void {
+    this.userService.getUsers().then((response: { content: any[]; }) => {
```

Comment:
Please make `getUsers` async and use `await` to wait for getUsers() and process the result in an apparently-synchronous way.
```ts
const response: { content: any[] } = await this.userService.getUsers();
// use response, without a .then() clause
```
`getUsers()` currently returns an Observable, convert it to a Promise by wrapping this function's return value with `lastValueFrom()` (eg. `lastValueFrom(this.http.get<..>(..)`) so that you can `await` it.